srcs/app/http/response/HttpResponseFactory.cpp

```diff
@@ -47,21 +48,27 @@ void	HttpResponseFactory::SetRequestConfig_() {
 }
 
 IResponse *HttpResponseFactory::createHttpRedirectionResponse_() {
-	return new HttpRedirectionResponse(request_config_, request_);
+	return Free_(new HttpRedirectionResponse(request_config_, request_));
 }
 
 IResponse *HttpResponseFactory::createHttpGetResponse_() {
-	return new HttpGetResponse(request_config_, request_);
+	return Free_(new HttpGetResponse(request_config_, request_));
 }
 
 IResponse *HttpResponseFactory::createHttpPostResponse_() {
-	return new HttpPostResponse(request_config_, request_);
+	return Free_(new HttpPostResponse(request_config_, request_));
 }
 
 IResponse *HttpResponseFactory::createHttpDeleteResponse_() {
-	return new HttpDeleteResponse(request_config_, request_);
+	return Free_(new HttpDeleteResponse(request_config_, request_));
 }
 
 IResponse *HttpResponseFactory::createHttpErrorResponse_(int statusCode) {
-	return new HttpErrorResponse(statusCode, request_config_, request_);
+	return Free_(new HttpErrorResponse(statusCode, request_config_, request_));
+}
+
+IResponse *HttpResponseFactory::Free_(IResponse *response) {
+	delete request_config_;
+	request_config_ = NULL;
+	return response;
 }
```

Comment:
Despite is a great idea just deleting the `request_config` at this point I don't feel comfortable with having that wrapper method because it make it less readable. What are you returning, a `Free`? what is a `Free`? (I understand what it is but, if the function is returning an `IResponse` you would expect that, a response) Having this wrapper seems kind of a hack.

I think, since the leak seems to come when de-referencing the pointer in `SetRequestConfig_()` method when receiving new messages on the same connection (correct me if I am wrong), to check if `request_config_` has content before allocating a new object. Something like: 

```c++
void	HttpResponseFactory::SetRequestConfig_() {
    if (request_config_) {
    	delete request_config_;
	request_config_ = NULL;
    }
    request_config_ = new RequestConfig(server_config_, request_->GetDecodedPath());
}
```

This way, and keeping deletion in the destructor should be fine.

Reply:
Victor, why do you want to check the non-nullity of request_config_? If it's not null, you'll always delete it. If it's null, you don't delete it, but you may as well delete it, since deleting a NULL pointer is completely safe.

Reply:
I like your idea :+1:. It's much cleaner now, thank you.

Reply:
You are totally right. I may just wanted to point out the reason of deleting it (when you had previous content) But I totally agree that we could get rid of the if statement.

Reply:
I've seen that, despite you said you liked the idea, with the new update there haven't been any changes. Do you plan on updating this?

Reply:
If I understand correctly, the suggestion was to remove the `Free_()` wrapper and `delete` the `request_config_` in the `SetRequestConfig_()`.

Is there anything wrong with how it is done in [this](https://github.com/42AGV/webserv/pull/102/commits/121096a37c207e5188509e484041be84eee20a93) commit?

Reply:
No, it is completely fine. I think I read it the other way, or even maybe an older version I have opened in another tab, Sorry and thanks :smile: